patches/api/0324-Add-critical-damage-API.patch

```diff
@@ -0,0 +1,111 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dodison <kacpik@mapik.eu>
+Date: Mon, 26 Jul 2021 17:35:20 +0200
+Subject: [PATCH] Add critical damage API
+
+
+diff --git a/src/main/java/org/bukkit/event/entity/EntityDamageByBlockEvent.java b/src/main/java/org/bukkit/event/entity/EntityDamageByBlockEvent.java
+index 461727dc7f06efb3550fc370e0db5bd04ba89711..3ee2610fdd5ec981421f4621bf91cf506390c35a 100644
+--- a/src/main/java/org/bukkit/event/entity/EntityDamageByBlockEvent.java
++++ b/src/main/java/org/bukkit/event/entity/EntityDamageByBlockEvent.java
+@@ -12,17 +12,22 @@ import org.jetbrains.annotations.Nullable;
+  */
+ public class EntityDamageByBlockEvent extends EntityDamageEvent {
+     private final Block damager;
+-
++    //Paper start - add critical damage API
```

Comment:
This Paper comment block is way bigger than it needs to be, Paper only really adds the last of these 3 constructors but having the block cover all three can be misleading. Add single-line paper comments for the `@ Deprecated` annotations, and thing paper comments for the last bit of changes where you actually implement the new constructor.

Example: https://github.com/PaperMC/Paper/pull/5638/files